docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1104 @@
+_uuid               : cd9264de-f2c6-4528-8f53-c2ccb5b23c8d
+allowed_ext_ips     : []
+exempted_ext_ips    : ["a15397791517117706455"] # Exclude traffic to the pod Network
+external_ids        : {}
+external_ip         : "172.18.0.2"         # Node/GR external IP used for SNAT
+external_mac        : []
+external_port_range : "32768-60999"
+gateway_port        : []
+logical_ip          : "10.128.0.7"         # The local pod IP
+logical_port        : []
+match               : ""
+options             : {stateless="false"}
+priority            : 0
+type                : snat
+```
+
+##### Local Gateway Mode
+
+In the following example, `10.128.0.0/24` is the pod subnet in the local node,
+`10.128.0.0/16` is the CIDR of the pod network. `natOutbound` is enabled in for
+the network.
+
+```text
+table ip nat {
+    chain postrouting {
+        type nat hook postrouting priority 100; policy accept;
+        # skip SNAT for pod-to-pod traffic
+        ip saddr 10.128.0.0/24 ip daddr 10.128.0.0/16 return
+        ip saddr 10.128.0.0/24 masquerade
+    }
+}
+```
+
+#### UDN Traffic Isolation
+
+Currently, for cross-UDN pod-to-pod traffic, OVN-Kubernetes supports 2 modes:
+loose and strict. In the loose mode, cross-UDN traffic is allowed if an external
+router can route the traffic correctly. In strict mode, cross-UDN traffic is
+blocked by ACL rule defined in the node switch. In no-overlay mode, these two
+isolation modes shall still operate in the same way.
+
+### Workflow
+
+#### Enable No-overlay Mode for the Default Network
+
+1. The frr-k8s pods shall be deployed on each node.
+2. Before starting the ovnkube pods, the cluster admin shall create a base
+   FRRConfiguration CR that is used for generating the per-node FRRConfiguration
+   instances by OVN-Kubernetes.
+
+    ```yaml
+    apiVersion: frrk8s.metallb.io/v1beta1
+    kind: FRRConfiguration
+    metadata:
+      name: default
+      namespace: frr-k8s-system
+      labels:
+        # A custom label
+        network: default
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          neighbors:
+          # the external BGP route reflector
+          - address: 172.20.0.2
+            asn: 64512
+            disableMP: true
+    ```
+
+3. The cluster admin shall create a RouteAdvertisements CR to advertise the
+   default network. Make sure the RouteAdvertisements CR selects the
+   FRRConfiguration defined in the previous step.
+
+    ```yaml
+    apiVersion: k8s.ovn.org/v1
+    kind: RouteAdvertisements
+    metadata:
+      name: default
+    spec:
+      advertisements:
+      - PodNetwork
+      # Select the FRRConfiguration defined in step-2 with the custom label.
+      frrConfigurationSelector:
+        matchLabels:
+          network: default
+      networkSelectors:
+      - networkSelectionType: DefaultNetwork
+      # The empty nodeSelector selects all nodes. We don't support a network in a overlay and no-overlay hybrid mode.
+      nodeSelector: {}
```

Comment:
Do we report an error somewhere if the nodeSelector isn't empty?

Reply:
ovnkube-cluster-manager shall print error to its log.